src/fromPairs.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Creates a new object from an array of tuples by pairing up first and second elements as {[key]: value}.
+ * If a tuple is not supplied for any element in the array, the element will be ignored
+ * If duplicate keys exist, the tuple with the greatest index in the input array will be preferred.
+ * @param tuples the list of input tuples
+ * @signature
+ *   R.fromPairs(tuples)
+ * @example
+ *   R.fromPairs([['a', 'b'], ['c', 'd']]) // => {a: 'b', c: 'd'}
+ * @category Object
+ */
+export function fromPairs<K extends string | number | symbol, V>(tuples: Array<[K, V]>): Record<K, V>
+
+
+export function fromPairs(tuples: Array<[string | number | symbol, unknown]>) {
+  return tuples.reduce((acc, curr) => curr && curr.length === 2 ? ({
+      ...acc,
```

Comment:
Can you mutate `acc` instead creating a copy?
If there is many elements, you can N copies and properties of the result object.

For example:
// n = 1000
// iteration 1 -> copy of object with 1 prop
// iteration  2 -> copy of object with 2 props
// iteration 3 -> copy of object with 3 props
...
// iteration 1000-> copy of object with 1000 props

1 + 2  + 3 + ... + n = n * (n + 1) / 2
= ~500,000